src/sema/using.rs

```diff
@@ -113,19 +116,145 @@ pub(crate) fn using_decl(
                         continue;
                     }
 
-                    if let Some(ty) = &ty {
-                        if *ty != func.params[0].ty {
+                    let oper = if let Some(mut oper) = using_function.oper {
+                        if contract_no.is_some() || using.global.is_none() || ty.is_none() {
+                            diagnostics.push(Diagnostic::error(
+                                using_function.loc,
+                                "user defined operator can only be set in a global 'using for' directive".into(),
+                            ));
+                            break;
+                        }
+
+                        let ty = ty.as_ref().unwrap();
+
+                        if !matches!(*ty, Type::UserType(_)) {
+                            diagnostics.push(Diagnostic::error(
+                                using_function.loc,
+                                format!("user defined operator can only be used with user defined types, type {} not permitted", ty.to_string(ns))
```

Comment:
```suggestion
                                format!("user defined operator can only be used with user defined types. Type {} not permitted", ty.to_string(ns))
```
That was a comma splice!